core/src/main/java/com/linecorp/armeria/client/endpoint/PropertiesFileWatcherRegistry.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.endpoint;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_DELETE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+import static java.nio.file.StandardWatchEventKinds.OVERFLOW;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+/**
+ * Wraps a WatchService and allows paths to be registered.
+ */
+final class PropertiesFileWatcherRegistry {
+    private static final Logger logger = LoggerFactory.getLogger(PropertiesFileWatcherRegistry.class);
+
+    @Nullable
+    private CompletableFuture<Void> future;
+
+    private final Map<String, PropertiesFileWatcherContext> ctxRegistry = new ConcurrentHashMap<>();
+    private final WatchService watchService;
+    private final ExecutorService eventLoop =
```

Comment:
We use `EventLoop` throughout the codebase with a very specific meaning (Netty event loop), it would be better to call this `executor` to prevent confusion.